scripts/test_ports/cmake-user/portfile.cmake

```
@@ -0,0 +1,159 @@
+            SHA512 455b8f940ccda0ba1169d3620db67c0bf89284126386408cd28b76b66c59c4c2ea5ad8def0095166e7524f6cf5202f117a2fa49e1525f93ed711657a5d2ae988
+        )
+        set(cmake_bin_dir "/bin")
+    else()
+        message(FATAL_ERROR "Unable to test feature 'cmake-3-4' for '${HOST_TRIPLET}' host.")
+    endif()
+
+    vcpkg_extract_source_archive_ex(
+        OUT_SOURCE_PATH legacy_cmake
+        ARCHIVE "${legacy_cmake_archive}"
+        REF "${cmake_version}"
+        WORKING_DIRECTORY "${CURRENT_BUILDTREES_DIR}/${name}"
+    )
+    set(CMAKE_COMMAND "${legacy_cmake}${cmake_bin_dir}/cmake")
+endif()
+
+set(packages "")
+if("find-package" IN_LIST FEATURES)
+    file(READ "${CMAKE_CURRENT_LIST_DIR}/vcpkg.json" vcpkg_json)
+    string(JSON packages_json GET "${vcpkg_json}" "features" "find-package" "dependencies")
+    string(JSON packages_count LENGTH "${packages_json}")
+    if(packages_count GREATER 0)
+        math(EXPR last "${packages_count} - 1")
+        foreach(i RANGE 0 ${last})
+            string(JSON package GET "${packages_json}" ${i} "$package")
+            list(APPEND packages "${package}")
+        endforeach()
+    endif()
+    if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
+        list(REMOVE_ITEM packages "Curses")
+    endif()
+endif()
+
+if(DEFINED ENV{VCPKG_FORCE_SYSTEM_BINARIES})
+    set(NINJA "ninja")
+else()
+    vcpkg_find_acquire_program(NINJA)
+endif()
+
+function(test_cmake_project)
+    cmake_parse_arguments(PARSE_ARGV 0 "arg" "" "NAME" "OPTIONS")
+    if(NOT arg_NAME)
+        message(FATAL_ERROR "The NAME argument is mandatory.")
+    endif()
+
+    set(build_dir "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-${arg_NAME}")
+    file(REMOVE_RECURSE "${build_dir}")
+    file(MAKE_DIRECTORY "${build_dir}")
+
+    if(DEFINED VCPKG_CMAKE_SYSTEM_NAME AND VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
+        list(APPEND arg_OPTIONS "-DCMAKE_SYSTEM_NAME=${VCPKG_CMAKE_SYSTEM_NAME}")
+        if(DEFINED VCPKG_CMAKE_SYSTEM_VERSION)
+            list(APPEND arg_OPTIONS "-DCMAKE_SYSTEM_VERSION=${VCPKG_CMAKE_SYSTEM_VERSION}")
+        endif()
+        if(DEFINED VCPKG_PLATFORM_TOOLSET)
+            list(APPEND arg_OPTIONS "-DVCPKG_PLATFORM_TOOLSET=${VCPKG_PLATFORM_TOOLSET}")
+        endif()
+    endif()
+    
+    if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+        list(APPEND arg_OPTIONS -DBUILD_SHARED_LIBS=ON)
+    else()
+        list(APPEND arg_OPTIONS -DBUILD_SHARED_LIBS=OFF)
+    endif()
+
+    vcpkg_execute_required_process(
+        COMMAND
+            "${CMAKE_COMMAND}" "${CMAKE_CURRENT_LIST_DIR}/project"
+            -G "Ninja"
+            "-DCMAKE_MAKE_PROGRAM=${NINJA}"
+            "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_CURRENT_LIST_DIR}/../../buildsystems/vcpkg.cmake"
+            "-DVCPKG_INSTALLED_DIR=${_VCPKG_INSTALLED_DIR}"
+            "-DCMAKE_INSTALL_PREFIX=${build_dir}/install"
+            "-DVCPKG_TARGET_TRIPLET=${TARGET_TRIPLET}"

```

We think we need to do something to avoid recursive `vcpkg` by force-disabling manifest mode here. (in case there is a vcpkg.json somewhere in a directory "above" this vcpkg working copy.)

```suggestion
            "-DVCPKG_TARGET_TRIPLET=${TARGET_TRIPLET}"
            "-DVCPKG_MANIFEST_MODE=OFF"

```

Ditto all other cmake invocations.